p2p/host.go

```diff
@@ -87,6 +87,19 @@ func NewHost(self *Peer, key libp2p_crypto.PrivKey) (Host, error) {
 		libp2p_pubsub.WithValidateThrottle(MaxMessageHandlers),
 	}
 
+	traceFile := os.Getenv("P2P_TRACEFILE")
+	if len(traceFile) > 0 {
+		pi, err := libp2p_peer.AddrInfoFromP2pAddr(ma.StringCast(traceFile))
+		if err == nil {
+			tracer, _ := libp2p_pubsub.NewRemoteTracer(ctx, p2pHost, *pi)
```

Comment:
It seems that the remote tracer will write all `TraceEvents` to the stream to peer `pi`. But it seems that the libp2p stream currently is not handled in our code. So is this something that we expect? 

```
~/go/pkg/mod/github.com/libp2p/go-libp2p-pubsub@v0.3.1/tracer.go:238
```

Do you think that a JSON tracer will be better for dumping the trace event to a local file?

```
~/go/pkg/mod/github.com/libp2p/go-libp2p-pubsub@v0.3.1/tracer.go:89
```

Reply:
we used to have local tracer actually.  We can support both though. a remote tracer is easier to collect all logs for us to analysis.   I'll add the support of both remote and local tracer then.